indra/explanation/cycle_free_paths.py

```diff
@@ -314,7 +314,9 @@ def _cf_succ(H, t, path, v):
     for u in H.successors(v):
         if set(path) <= set(t[u]):
             succ.append(u)
-    w = random.choice(succ)
+    idx_list = list(range(len(succ)))
+    w_idx = np.random.choice(idx_list)
+    w = succ[w_idx]

```

Comment:
Are these 3 lines necessary? Wouldn't just using ```w = np.random.choice(succ)``` be the same?

Reply:
You'd think, but that's to avoid the weird way np.random handles lists of
lists (it raises an exception). If you're just trying to pick a random
list, you have to specify that by creating an index and choosing from that.
You can actually see basically the same technique used in the path graph, I
think. Or perhaps it was a different instance of random.